relay_client/src/error.rs

```diff
@@ -42,15 +44,69 @@ pub enum Error {
     #[error("Invalid response ID")]
     InvalidResponseId,
 
+    #[error("Invalid error response")]
+    InvalidErrorResponse,
+
     #[error("Serialization failed: {0}")]
     Serialization(serde_json::Error),
 
     #[error("Deserialization failed: {0}")]
     Deserialization(serde_json::Error),
 
-    #[error("RPC error ({code}): {message}")]
-    Rpc { code: i32, message: String },
+    #[error("RPC error: code={code} data={data:?} message={message}")]
+    Rpc {
+        code: i32,
+        message: String,
+        data: Option<String>,
+    },
 
     #[error("Invalid request type")]
     InvalidRequestType,
 }
+
+impl From<rpc::ErrorData> for ClientError {
+    fn from(err: rpc::ErrorData) -> Self {
+        Self::Rpc {
+            code: err.code,
+            message: err.message,
+            data: err.data,
+        }
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error<T: ServiceError> {
```

Comment:
Do we need to have this bound on the struct itself? If not, I suggest removing it, it generally makes the life harder for the one who uses the type